pulpcore/pulpcore/app/models/publication.py

```diff
@@ -20,6 +42,68 @@ class Publication(Model):
 
     repository_version = models.ForeignKey('RepositoryVersion', on_delete=models.CASCADE)
 
+    complete = False  # placeholder
+
+    @classmethod
+    def create(cls, repository_version, publisher):
+        """
+        Create a publication.
+
+        This should be used to create a publication.  Using Publication() directly
+        is highly discouraged.
+
+        Args:
+            repository_version (pulpcore.app.models.RepositoryVersion): The repository
+                version to be published.
+            publisher (pulpcore.app.models.Publisher): The publisher used
+                to create the publication.
+
+        Returns:
+            pulpcore.app.models.Publication: self
+
+        Notes:
+            The created publication with complete=False.
+            Adds a Task.created_resource for the publication.
+        """
+        with transaction.atomic():
+            publication = cls(
+                repository_version=repository_version,
+                publisher=publisher)
+            publication.save()
+            resource = CreatedResource(content_object=publication)
+            resource.save()
+            return publication
+
+    def delete(self, **kwargs):
+        """
+        Delete the publication.
+
+        Args:
+            **kwargs (dict): Delete options.
+
+        Notes:
+            Deletes the Task.created_resource when incomplete.
+        """
+        with transaction.atomic():
+            if not self.complete:
+                CreatedResource.objects.delete(object_id=self.pk)
```

Comment:
I think we should delete the CreatedResource even if the Publication was never marked complete.

Reply:
👍